src/Security/Authentication/Facebook/src/FacebookHandler.cs

```diff
@@ -64,7 +64,7 @@ private string GenerateAppSecretProof(string accessToken)
         var builder = new StringBuilder();
         for (int i = 0; i < hash.Length; i++)
         {
-            builder.Append(hash[i].ToString("x2", CultureInfo.InvariantCulture));
+            builder.Append(CultureInfo.InvariantCulture, $"{hash[i]:x2}");
```

Comment:
FYI @martincostello and @javiercn in regards to the conversation at https://github.com/dotnet/aspnetcore/pull/36368/files#r706039158

This makes use of [Interpolated string handlers](https://learn.microsoft.com/dotnet/csharp/whats-new/tutorials/interpolated-string-handler), and `StringBuilder` has a custom one which allows formatting the interpolated string directly into it's internal buffer instead of instantiating a string then copying the string.

|             Method |     Mean |    Error |   StdDev |   Gen0 | Allocated |
|------------------- |---------:|---------:|---------:|-------:|----------:|
|     AppendToString | 748.7 ns | 12.33 ns | 10.93 ns | 0.1841 |    1448 B |
| AppendInterpolated | 739.7 ns |  9.29 ns |  8.23 ns | 0.0620 |     488 B |

<details>
<summary>Benchmark code</summary>

```csharp
[MemoryDiagnoser]
public class AppendBenchmark
{
    private byte[] _b = new byte[30];

    [GlobalSetup]
    public void Setup()
    {
        RandomNumberGenerator.Fill(_b);
    }

    [Benchmark]
    public string AppendToString()
    {
        var sb = new StringBuilder();
        foreach (var b in _b)
        {
            sb.Append(b.ToString("x2", CultureInfo.InvariantCulture));
        }
        return sb.ToString();
    }

    [Benchmark]
    public string AppendInterpolated()
    {
        var sb = new StringBuilder();
        foreach (var b in _b)
        {
            sb.Append(CultureInfo.InvariantCulture, $"{b:x2}");
        }
        return sb.ToString();
    }
}
```

</details>

Reply:
Nice 😎